src/Campaigns/Repositories/CampaignRepository.php

```diff
@@ -138,6 +140,50 @@ public function update(Campaign $campaign): void
         Hooks::doAction('givewp_campaign_updated', $campaign);
     }
 
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function addCampaignForm(Campaign $campaign, DonationForm $donationForm, bool $isDefault = false)
```

Comment:
Do we need a Donation form object here or do we need just the form ID? I don't see the benefit of passing the entire object where just an ID will be sufficient.

Reply:
This way we don't need to use the `Campaign::find()` method to check if the campaign exists. If we already have the object, we know that the campaign exists.

Reply:
> Do we need a Donation form object here or do we need just the form ID?

I was talking about `DonationForm`.

Reply:
We can apply the same logic as we don't want to add a form id that doesn't exist on the DB.

Reply:
Having a `DonationForm` object here doesn't necessarily mean that form exists, but okay.